pkg/operator/ceph/object/user/controller.go

```diff
@@ -472,13 +478,16 @@ func generateStatusInfo(u *cephv1.CephObjectStoreUser) map[string]string {
 	return m
 }
 
-func (r *ReconcileObjectStoreUser) generateCephUserSecret(u *cephv1.CephObjectStoreUser) *corev1.Secret {
+func (r *ReconcileObjectStoreUser) generateCephUserSecret(u *cephv1.CephObjectStoreUser, tlsSecretName string) *corev1.Secret {
 	// Store the keys in a secret
 	secrets := map[string]string{
 		"AccessKey": r.userConfig.Keys[0].AccessKey,
 		"SecretKey": r.userConfig.Keys[0].SecretKey,
 		"Endpoint":  r.objContext.Endpoint,
 	}
+	if tlsSecretName != "" {
+		secrets["SSLCertRef"] = tlsSecretName
```

Comment:
What format will the secret name be in? The secret is part of the Rook cluster namespace, so just the secret name might not be useful here. What about including the namespace in `NAMESPACE/SECRET_NAME` format?

Due to RBAC, most applications service accounts wouldn't be able to access the secret in the Rook cluster namespace anyway. Doesn't the endpoint/ URL change accordingly if SSL is enabled or not with `http`/ `https`?

Reply:
We can pass the entire cert as well, what do you think is optimal for sharing the cert directly or pass k8s secret with namespace?

If the Secure port is enable in CephObjectStore CRD, then the endpoint passed in CephObjectStoreUser secret will be https

Reply:
@travisn Yes, that's correct. Currently `CephObjectStoreUser` and its secret can only be created where `CephObjectStore` resides where the tls secret is also present